cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -73,12 +71,51 @@ func preRunCreateCluster(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func (cc *createClusterOptions) validate(ctx context.Context) error {
-	clusterConfig, err := commonValidation(ctx, cc.fileName)
+func (cc *createClusterOptions) createCluster(cmd *cobra.Command) error {
+	ctx := cmd.Context()
+
+	clusterConfigFileExist := validations.FileExists(cc.fileName)
+	if !clusterConfigFileExist {
+		return fmt.Errorf("the cluster config file %s does not exist", cc.fileName)
+	}
+
+	clusterConfig, err := v1alpha1.GetAndValidateClusterConfig(cc.fileName)
 	if err != nil {
-		return err
+		return fmt.Errorf("the cluster config file provided is invalid: %v", err)
+	}
+
+	if clusterConfig.Spec.DatacenterRef.Kind == v1alpha1.TinkerbellDatacenterKind {
+		flag := cmd.Flags().Lookup("hardwarefile")
+
+		// If no flag was returned there is a developer error as the flag has been removed
+		// from the program.
+		if flag == nil {
+			panic("'hardwarefile' flag not configured")
```

Comment:
why panic instead of returning an error here?

Reply:
If the flag is missing and we expect it there's a programmer error. I.e the program is invalid which is cause for a panic. To paint it with a different perspective, the user can't do anything about this problem. The binary is totally useless to them.